### PR TITLE
fix: add semantic versioning plugin

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes
Add semantic versioning plugin that was removed during gradle kts migration

#### Motivation and Context
This plugin is needed to augment project version during release phase

#### How Has This Been Tested?
Locally running `./gradlew incrementVersion --versionIncrementType=MAJOR` command and checking the effective version increment

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -98,6 +95,11 @@ sourceSets {
 
 springBoot { mainClass.set("it.pagopa.wallet.WalletApplicationKt") }
 
+tasks.create("applySemanticVersionPlugin") {
+  dependsOn("prepareKotlinBuildScriptModel")
+  apply(plugin = "com.dipien.semantic-version")
+}
+
```
Comment (giovanniberti) [inline build.gradle.kts:102]: Why is this needed?

Comment (pietro-tota) [reply build.gradle.kts:102]: Because plugin listed into plugin section are automatically applied, but by then the version is not already set making the plugin fail blaming about version to be set in root project.
I then set to false plugin apply into plugins section and apply it later after all the project variables have been set

**Review by giovanniberti — APPROVED** [review]
